### DISCUSSION TITLE
proper way to handle dynamic properties from JS code

### DISCUSSION THREAD

**Discussion by SheetJSDev** [denoland/deno#13652] [Q&A] [answered]
```js
// test.js
const x = {x: 2};
x.y = 3;
export { x };

// app.ts
import * as x from './test.js';
console.log(x.x.x);
console.log(x.x.y);
```

Deno fails with

```
error: TS2339 [ERROR]: Property 'y' does not exist on type '{ x: number; }'.
console.log(x.x.y);
                ^
```

In cases like this, where the object is modified, is there a way to hint from the JS code (no TS definition file) that the variable should be typed as `any` or `object`?

**Comment by kitsonk** [comment, marked as answer]
This really isn't a Deno specific question or problem. This would be the behaviour of any TypeScript project running in strict mode with `allowJs` enabled, which is the default for Deno TypeScript type checking.

You can use [JSDoc](https://www.typescriptlang.org/docs/handbook/jsdoc-supported-types.html) annotations in JavaScript to assign types:

```js
/** @type {any} */
const x = {x: 2};
x.y = 3;
export { x };
```
